orttraining/orttraining/python/training/_ortmodule_logger.py

```diff
@@ -0,0 +1,51 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+
+from contextlib import contextmanager
+from enum import IntEnum
+import io
+import sys
+import tempfile
+import warnings
+
+class LogLevel(IntEnum):
+    VERBOSE = 0
+    INFO = 1
+    WARNING = 2
+    ERROR = 3
+    FATAL = 4
+
+
+@contextmanager 
+def suppress_os_stream_output(suppress_stdout=True, suppress_stderr=True, log_level=LogLevel.WARNING):
+    """Supress output from being printed to stdout and stderr if log_level is WARNING or higher.
+
+    If there is any output detected, a single warning is issued at of the context
+    """
+
+    # stdout and stderr is written to a tempfile instead
+    stdout = sys.stdout
+    stderr = sys.stderr
+
+    suppress_logs = log_level >= LogLevel.WARNING
+
+    with tempfile.TemporaryFile(mode='w') as fo:
```

Comment:
Does it work to create a StringIO object here instead of a literal file-on-disk?

Reply:
Yes, that is a good idea. Thanks.